src/arch/s390.cpp

```diff
@@ -90,10 +89,9 @@ std::string name()
   return std::string("s390x");
 }
 
-std::vector<std::string> invalid_watchpoint_modes()
+bool is_watchpoint_mode_valid(const std::string &mode)
 {
-  throw FatalUserException(
```

Comment:
Not that this was the correct place for this but should we add a check in one of the passes to error that watchpoints are not supported on X architecture? It seems users will now get an error "invalid watchpoint mode", which is not correct.

Reply:
I can restructure further to return the list of valid watchpoints, and add a hint.